### PR TITLE
table_wrapper.rs: Fix for percentage widths

### LINKED ISSUE #4421: > 100% width on table mishandled
The following should be equivalent:
```html
<!doctype html>
<title>Test for capping percentages</title>
<style>
div { width:300px; background:yellow; height:50px; }
table { width:150%; }
td { background:blue; }
</style>
<div>
  <table cellspacing="0" cellpadding="0" border="0">
    <tr><td>parent div float=left</td></tr>
  </table>
</div>
```
```html
<!doctype html>
<title>Test for capping percentages</title>
<style>
div { width:300px; background:yellow; height:50px; }
table { width:450px; }
td { background:blue; }
</style>
<div>
  <table cellspacing="0" cellpadding="0" border="0">
    <tr><td>parent div float=left</td></tr>
  </table>
</div>
```

### PR DESCRIPTION
**Pull request by mttr**
Fixes #4421

### CODE REVIEW

**Comment by jdm** [comment]
Note: I know nothing about table layout and I didn't read the changes. Is this change covered by an existing spec?

**Comment by mttr** [comment]
I changed some of the existing code to match with the "Column and caption widths influence the final table width as follows: ..."  part toward the end of [this section](http://dev.w3.org/csswg/css2/tables.html#auto-table-layout) of the CSS spec (the end result should be, as the spec describes [here](http://dev.w3.org/csswg/css2/tables.html#width-layout) that "if the margins of a table are set to '0' and the width to 'auto', the table will not automatically size to fill its containing block. However, once the calculated value of 'width' for the table is found (using the algorithms given below or, when appropriate, some other UA dependent algorithm) then the other parts of section 10.3 do apply").

I'm not 100% confident about my changes since I can't claim to fully understand the layout code, but it seems to work (and I included a new reftest based on the tests in #4421).

File: components/layout/table_wrapper.rs [hunk 1]
```diff
@@ -128,8 +128,18 @@ impl TableWrapperFlow {
         // FIXME(pcwalton, spec): INTRINSIC § 8 does not properly define how to compute this, but
         // says "the basic idea is the same as the shrink-to-fit width that CSS2.1 defines". So we
         // just use the shrink-to-fit inline size.
-        let available_inline_size =
-            self.block_flow.get_shrink_to_fit_inline_size(available_inline_size);
+        let available_inline_size = match self.block_flow.fragment.style().content_inline_size() {
+            LengthOrPercentageOrAuto::Auto => self.block_flow
+                                                  .get_shrink_to_fit_inline_size(available_inline_size),
+            // XXX(mttr) This fixes #4421 without breaking our current reftests, but I'm
```
Comment (pcwalton) [inline components/layout/table_wrapper.rs]: nit: I prefer `FIXME` over `XXX`

**Comment by pcwalton** [comment]
Looks good modulo that one nit!

**Comment by pcwalton** [comment]
LGTM, squash the two into one and I'll r+. Thanks!

**Comment by mttr** [comment]
Got it!

**Pull request merged**